components/ModalEdit.js

```diff
@@ -0,0 +1,266 @@
+import { useState } from "react";
+import styled from "styled-components";
+import { uid } from "uid";
+
+export default function ModalEdit({ currentProject, onSave, onCancel }) {
+  const [updateProject, setUpdateProject] = useState(currentProject);
+  const [selectSteps, setSelectSteps] = useState({});
+  const [updateSteps, setUpdateSteps] = useState(false);
+  const [updateMaterials, setUpdateMaterials] = useState(
+    currentProject.materials.join(", ")
+  );
+  const complexityLevels = ["Beginner", "Intermediate", "Advanced"];
+
+  function handleChange(event) {
+    const materialName = event.target.name;
+    const materialValue = event.target.value;
+
+    if (materialName === "materials") {
+      setUpdateMaterials(materialValue);
+      setUpdateProject((prevProject) => ({
+        ...prevProject,
+        [materialName]: materialValue
+          .split(",")
+          .map((material) => material.trim()),
+      }));
+    } else {
+      setUpdateProject((prevProject) => ({
+        ...prevProject,
+        [materialName]: materialValue,
+      }));
+    }
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    onSave(updateProject);
+  }
+
+  function handleStepAdd(event) {
+    event.preventDefault();
+    setUpdateProject((prevState) => ({
+      ...prevState,
+      steps: [...prevState.steps, { id: uid(), desc: "" }],
+    }));
+  }
+
+  function handleStepDelete(event) {
+    event.preventDefault();
+    setUpdateProject((prevState) => ({
+      ...prevState,
+      steps: prevState.steps.filter((step) => !selectSteps[step.id]),
+    }));
+    setSelectSteps({});
+  }
+
+  function handleStepEdit(event) {
+    event.preventDefault();
+    setUpdateSteps(true);
+  }
+
+  function handleStepEditCancel(event) {
+    event.preventDefault();
+    setUpdateSteps(false);
+  }
+
+  function handleStepSelect(id) {
+    setSelectSteps((prevState) => ({
+      ...prevState,
+      [id]: !prevState[id],
+    }));
+  }
+
+  return (
+    <StyledBackground>
+      <StyledContainer>
+        <StyledContainerForm>
+          <form onSubmit={handleSubmit}>
+            <StyledHeading>Edit Project</StyledHeading>
+            <label htmlFor="title">
+              <StyledTitle>Title</StyledTitle>
```

Comment:
```suggestion
            <label htmlFor="title">Title</label>
```

Reply:
Thank you for the input, for now we'll keep it since this is just a really basic styling/layout for having things aligned initially.